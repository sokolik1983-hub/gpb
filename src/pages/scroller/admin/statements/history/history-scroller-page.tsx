import React from 'react';
import type { ScrollerPageLayoutProps } from 'components/admin';
import { SCROLLER_PAGE_LAYOUT_HEADER_HEIGHT, ScrollerPageLayout } from 'components/admin';
import { FocusLock } from 'components/common/focus-lock';
import { FocusTree } from 'components/common/focus-tree';
import { useStreamContentHeight } from 'hooks/common';
import { locale } from 'localization';
import { Table } from 'pages/scroller/admin/statements/components';
import { Filter } from 'pages/scroller/admin/statements/components/filter';
import { FILTER_STORAGE_KEY, TABLE_STORAGE_KEY } from 'pages/scroller/admin/statements/history/constants';
import { useHeaderActions, useScrollerData } from 'pages/scroller/admin/statements/hooks';
import { statementService } from 'services/admin';
import { QUICK_FILTER_HEIGHT } from 'stream-constants';
import { COMMON_SCROLLER_NODE } from 'stream-constants/a11y-nodes';
import { MainLayout } from '@platform/services/admin';

/**
 * [Выписки_ЗВ] ЭФ Банка "Журнал запросов выписки".
 *
 * @see https://confluence.gboteam.ru/pages/viewpage.action?pageId=32247657
 */
export const StatementHistoryScrollerPage = () => {
  const { datePeriodInitialed, filter, setFilter, setDatePeriodFetched, setStatements, statements } = useScrollerData();

  const actions = useHeaderActions({
    dateFrom: filter.dateFrom?.value,
    dateTo: filter.dateTo?.value,
    statements,
  });

  const headerProps: ScrollerPageLayoutProps['headerProps'] = {
    actions,
    header: locale.admin.historyScroller.pageTitle,
  };

  const height = useStreamContentHeight();

  return (
    <MainLayout>
      <FocusLock>
        <FocusTree treeId={COMMON_SCROLLER_NODE}>
          <ScrollerPageLayout headerProps={headerProps}>
            <Filter setDatePeriodFetched={setDatePeriodFetched} setFilter={setFilter} storageKey={FILTER_STORAGE_KEY} />
            <Table
              apiMethod={statementService.getStatementHistoryList}
              filter={filter}
              height={height - SCROLLER_PAGE_LAYOUT_HEADER_HEIGHT.BASE - QUICK_FILTER_HEIGHT}
              setStatements={setStatements}
              show={datePeriodInitialed}
              storageKey={TABLE_STORAGE_KEY}
            />
          </ScrollerPageLayout>
        </FocusTree>
      </FocusLock>
    </MainLayout>
  );
};

StatementHistoryScrollerPage.displayName = 'StatementHistoryScrollerPage';
