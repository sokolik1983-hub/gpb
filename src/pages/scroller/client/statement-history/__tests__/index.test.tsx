import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import { createMemoryHistory } from 'history';
import { Router } from 'react-router-dom';
import { StatementHistoryScrollerPage } from '..';

describe('StatementHistoryScrollerPage', () => {
  it('renders', () => {
    // Подход взят отсюда https://testing-library.com/docs/example-react-router/
    const history = createMemoryHistory();

    const wrapper = shallow(
      <Router history={history}>
        <StatementHistoryScrollerPage />
      </Router>
    );

    // wrapper.find(StatementHistoryScrollerPage) используется, чтобы решить эту https://stackoverflow.com/q/44490100/13262779 проблему
    expect(toJson(wrapper.find(StatementHistoryScrollerPage))).toMatchSnapshot();
  });
});
