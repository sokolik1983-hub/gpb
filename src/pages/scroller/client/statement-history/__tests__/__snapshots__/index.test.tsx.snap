// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StatementHistoryScrollerPage renders 1`] = `<StatementHistoryScrollerPage />`;
